### PR TITLE
Srcs 154 i2c controller

### PR DESCRIPTION
**Pull request by OliverNyholm**

### CODE REVIEW

File: Compass/HMC6343.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ HMC6343::HMC6343(unsigned int headningBufferSize) :
 	m_magX = 0;
 	m_magY = 0;
 	m_magZ = 0;
+	m_temperature = 0;
```
Comment (simchanu29) [inline Compass/HMC6343.cpp:19]: Add a comment : to disable a warning.

File: i2ccontroller/I2CController.cpp [hunk 2]
```diff
@@ -1,95 +1,94 @@
...
 
-I2CController::~I2CController() {
+int I2CController::setup(int address)
+{
+	if (locked)
+	{
+		return wiringPiI2CSetup(address);
+	}
 
+	return 0; //This should never happen unless you are silly. beginTransmission needs to be called before this function
```
Comment (simchanu29) [inline i2ccontroller/I2CController.cpp]: Comment on 2 lines would be better.

File: i2ccontroller/I2CController.cpp [hunk 3]
```diff
@@ -1,95 +1,94 @@
...
+		}
+		delay(1);
 
-    // NOTE - Jordan - I think the I2C device code should interact via the I2C controller that way we can make it thread safe.
+		return returnVector;
+	}
 
-    initCompass(m_mockCompass, m_headingBufferSize);
-    initArduino(m_mockArduino);
+	return {}; //This should never happen unless you are silly. beginTransmission needs to be called before this function
```
Comment (simchanu29) [inline i2ccontroller/I2CController.cpp]: Comment on 2 lines would be better

File: i2ccontroller/I2CController.cpp [hunk 4]
```diff
@@ -1,95 +1,94 @@
...
-        m_timer.sleepUntil(m_loopTime);
-    }
+int I2CController::write(int data)
+{
+	if (locked)
+	{
+		return wiringPiI2CWrite(m_fd, data);
+	}
+
+	return 0; //This should never happen unless you are silly. beginTransmission needs to be called before this function
```
Comment (simchanu29) [inline i2ccontroller/I2CController.cpp]: Comment on 2 lines would be better

File: i2ccontroller/I2CController.cpp [hunk 5]
```diff
@@ -1,95 +1,94 @@
...
-        return false;
-    }
-
-    Logger::info("Arduino initialised");
-    return true;
+int I2CController::readBlock()
+{
+	if(locked)
+	{
+		uint8_t block[40];
```
Comment (simchanu29) [inline i2ccontroller/I2CController.cpp]: Why is it a size of 40 ?

File: i2ccontroller/I2CController.cpp [hunk 6]
```diff
@@ -1,95 +1,94 @@
...
+		wiringPiI2CReadBlock(m_fd, block);
+
+		return block[10];
```
Comment (simchanu29) [inline i2ccontroller/I2CController.cpp]: Why do you return only the 10th value ?

File: i2ccontroller/I2CController.cpp [hunk 7]
```diff
@@ -1,95 +1,94 @@
...
+	locked = true;
 }
 
-void I2CController::close() {
-    m_mutex.lock();
-    m_running = false;
-    m_mutex.unlock();
+void I2CController::endTransmission()
+{
+	locked = false;
```
Comment (simchanu29) [inline i2ccontroller/I2CController.cpp]: This is a void function, so is locked a class variable ? If yes, why it's not named m_locked ?

File: i2ccontroller/I2CController.h [hunk 8]
```diff
@@ -1,47 +1,34 @@
...
-#include "thread/SystemState.h"
-#include "utility/Timer.h"
-
-#include <chrono>
-#include <thread>
+
+#include <vector>
+#include <stdint.h>
+#include <wiringPiI2C.h>
+#include <wiringPi.h> // delay
```
Comment (simchanu29) [inline i2ccontroller/I2CController.h]: What does that comment mean ? Does WiringPi takes care of some delays of some sort ?

Comment (JSpringett) [inline i2ccontroller/I2CController.h]: Usually it implies that that header is only used for the function delay, gives purpose behind the include.

Comment (Elessog) [inline i2ccontroller/I2CController.h]: the wiring includes should be in i2ccontroller.cpp because we don't need them
in the .h

Comment (JSpringett) [inline i2ccontroller/I2CController.h]: fixed

**Comment by Elessog** [comment]
approved

**Pull request merged**